src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.4.1.0.cs

```diff
@@ -82,25 +82,54 @@ public static void ServiceRestart_Throws_CommunicationException()
             ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)setupHostServiceProxy, setupHostFactory);
         }
 
-        // *** Second Part *** \\
-        CommunicationException exception = Assert.Throws<CommunicationException>(() =>
+        // *** Additional Setup *** \\
+        // The restartServiceAddress we got from the Service used localhost as the host name.
+        // We need the actual host name for the client call to work.
+        // To make it easier to parse, localhost was replaced with '[HOST]'.
+
+        // Use Endpoints.HttpBaseAddress_Basic only for the purpose of extracting the Service host name.
+        // Then update 'restartServiceAddress' with it.
+        UriBuilder builder = new UriBuilder(Endpoints.HttpBaseAddress_Basic);
```

Comment:
I believe this was existing code, but the use of UriBuilder is overkill. You can use `new Uri(Endpoints.HttpBaseAddress_Basic).Host`. UriBuilder is used when you want to modify a Uri without having to do any parsing.